packages/perseus/src/widgets/radio/base-radio.jsx

```diff
@@ -135,40 +144,56 @@ class BaseRadio extends React.Component<Props> {
     // will usually cause the old answer to become unchecked.)
     //
     // So, given the new values for a particular choice, compute the new values
-    // for all choices, and pass them to `this.props.onChange`.
+    // for all choices, and pass them to `onChange`.
     //
     // `newValues` is an object with two keys: `checked` and `crossedOut`. Each
     // contains a boolean value specifying the new checked and crossed-out
     // value of this choice.
-    updateChoice: (
-        number,
+    function updateChoice(
+        choiceIndex: number,
         newValues: $ReadOnly<{checked?: boolean, crossedOut?: boolean}>,
-    ) => void = (choiceIndex, newValues) => {
+    ): void {
+        const {multipleSelect, choices, onChange} = props;
+
         // Get the baseline `checked` values. If we're checking a new answer
         // and multiple-select is not on, we should clear all choices to be
         // unchecked. Otherwise, we should copy the old checked values.
         let newCheckedList;
-        if (newValues.checked && !this.props.multipleSelect) {
-            newCheckedList = this.props.choices.map((_) => false);
+        if (newValues.checked && !multipleSelect) {
+            newCheckedList = choices.map((_) => false);
         } else {
-            newCheckedList = this.props.choices.map((c) => c.checked);
+            newCheckedList = choices.map((c) => c.checked);
         }
 
         // Get the baseline `crossedOut` values.
-        const newCrossedOutList = this.props.choices.map((c) => c.crossedOut);
+        const newCrossedOutList = choices.map((c) => c.crossedOut);
 
         // Update this choice's `checked` and `crossedOut` values.
         newCheckedList[choiceIndex] = newValues.checked;
         newCrossedOutList[choiceIndex] = newValues.crossedOut;
 
-        this.props.onChange({
+        onChange({
             checked: newCheckedList,
             crossedOut: newCrossedOutList,
         });
-    };
+    }
 
-    focus: (number) => boolean = (i) => {
-        const ref = this.choiceRefs[i || 0];
+    const {
+        apiOptions,
+        reviewModeRubric,
+        choices,
+        editMode,
+        multipleSelect,
+        labelWrap,
+        countChoices,
+        numCorrect,
+        registerFocusFunction,
+    } = props;
+
+    // register a callback with the parent that allows
+    // the parent to focus an individual choice
+    registerFocusFunction?.((i) => {
```

Comment:
There's another place where BaseRadio was being used and the focus function was not being called (`widgets/radio/editor.jsx`). Rather than pass a dummy function I thought it would be simpler to make the registerFocusFunction optional.